internal/cli/cli.go

```diff
@@ -111,16 +113,38 @@ func (c *banzaiContext) SetOrganizationID(id int32) {
 func (c *banzaiContext) save() {
 	log.Debug("writing config")
 
-	if err := viper.WriteConfig(); err != nil {
-		log.Infof("failed to write config: %v", err)
+	if viper.ConfigFileUsed() == "" {
+		log.Debug("no config file defined, falling back to default location $HOME/.banzai")
+
 		home, _ := homedir.Dir()
 		configPath := path.Join(home, ".banzai")
-		os.MkdirAll(configPath, os.ModePerm)
+		err := os.MkdirAll(configPath, os.ModePerm)
+		if err != nil {
+			log.Fatal(emperror.Wrap(err, "failed to create config dir"))
+		}
+
 		configPath = path.Join(configPath, "config.yaml")
-		if err := viper.WriteConfigAs(configPath); err != nil {
+		err = viper.WriteConfigAs(configPath)
+		if err != nil {
 			log.Fatal(emperror.Wrap(err, "failed to write config"))
-		} else {
-			log.Infof("config created at %v", configPath)
 		}
+
+		log.Infof("config created at %v", configPath)
+		return
+	}
+
+	if _, err := os.Stat(filepath.Dir(viper.ConfigFileUsed())); os.IsNotExist(err) {
+		log.Debug("creating config dir")
+
+		configPath := filepath.Dir(viper.ConfigFileUsed())
+		err := os.MkdirAll(configPath, os.ModePerm)
```

Comment:
If config file will contain sensitive information (token) more strict directory permission should be used. (0700)